qcodes/instrument/base.py

```diff
@@ -114,24 +118,31 @@ def add_parameter(self, name: str,
                 unit of the new parameter is inconsistent with the existing
                 one.
         """
-        existing_parameter = self.parameters.get(name, None)
-
-        if existing_parameter:
+        if "bind_to_instrument" not in kwargs.keys():
+            kwargs["bind_to_instrument"] = True
 
-            if not existing_parameter.abstract:
-                raise KeyError(f"Duplicate parameter name {name}")
-
-            existing_unit = getattr(existing_parameter, "unit", None)
-            new_unit = kwargs.get("unit", None)
-            if existing_unit != new_unit:
-                raise ValueError(
-                    f"The unit of the parameter '{name}' is '{new_unit}'. "
-                    f"This is inconsistent with the unit defined in the "
-                    f"base class"
-                )
+        try:
+            param = parameter_class(name=name, instrument=self, **kwargs)
+        except TypeError:
+            kwargs.pop("bind_to_instrument")
+            warnings.warn(
+                f"Parameter {name} on instrument {self.name} does "
+                f"not correctly pass kwargs to its baseclass. A "
+                f"Parameter class must take `**kwargs` and forward "
+                f"them to its baseclass.",
+                QCoDeSDeprecationWarning,
+            )
+            param = parameter_class(name=name, instrument=self, **kwargs)
 
-        param = parameter_class(name=name, instrument=self, **kwargs)
-        self.parameters[name] = param
+        existing_parameter = self.parameters.get(name, None)
+        if not existing_parameter:
+            warnings.warn(
+                f"Parameter {name} does not correctly register on instrument"
+                f" {self.name}. Please check that instrument argument is passed "
+                f"to '_BaseParameter'. This will be an error in the future.",
```

Comment:
is the correction correct, or am i misunderstanding something?
```suggestion
                f"from {parameter_class!r} all the way to `_BaseParameter`. "
                "This will be an error in the future.",
```